x-pack/plugins/licensing/public/plugin.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { BehaviorSubject, Subject, merge, of, timer } from 'rxjs';
+import { filter, map, pairwise, switchMap, takeUntil, tap } from 'rxjs/operators';
+import {
+  CoreSetup,
+  CoreStart,
+  Plugin as CorePlugin,
+  PluginInitializerContext,
+} from 'src/core/public';
+import { LicensingPluginSetup, ILicensingPlugin, ObjectifiedLicense } from '../common/types';
+import {
+  API_ROUTE,
+  LICENSING_SESSION,
+  LICENSING_SESSION_SIGNATURE,
+  SIGNATURE_HEADER,
+  DEFAULT_POLLING_FREQUENCY,
+} from '../common/constants';
+import { License } from '../common/license';
+import { hasLicenseInfoChanged } from '../common/has_license_info_changed';
+
+export class Plugin implements CorePlugin<LicensingPluginSetup>, ILicensingPlugin {
+  private stop$ = new Subject();
+  private removeInterceptor!: () => void;
+  private signature = '';
+  public pollingFrequency = DEFAULT_POLLING_FREQUENCY;
```

Comment:
users may set another value in the config, so we shouldn't read default one.

Reply:
There is no config on the client-side, only changing the property prior to setup. This means the default one should be read until overwritten.

Reply:
the name doesn't tell that we allow users to specify only server side

Reply:
I'm confused why we would capture this in the name of the variable. I will set this in the comments for sure. :+1: